core/dbt/parser/manifest.py

```diff
@@ -270,20 +274,17 @@ def load(self):
             # We need to parse the macros first, so they're resolvable when
             # the other files are loaded
             start_load_macros = time.perf_counter()
-            for project in self.all_projects.values():
-                if project.project_name not in project_parser_files:
-                    continue
-                parser_files = project_parser_files[project.project_name]
-                if 'MacroParser' not in parser_files:
-                    continue
-                parser = MacroParser(project, self.manifest)
-                for file_id in parser_files['MacroParser']:
-                    block = FileBlock(self.manifest.files[file_id])
-                    parser.parse_file(block)
-                    # increment parsed path count for performance tracking
-                    self._perf_info.parsed_path_count = self._perf_info.parsed_path_count + 1
-            # Look at changed macros and update the macro.depends_on.macros
-            self.macro_depends_on()
+            self.load_and_parse_macros(project_parser_files)
+
+            # If we're partially parsing check that certain macros have not been changed
+            if self.partially_parsing and self.skip_partial_parsing_because_of_macros():
+                logger.info("Partial parsing enabled, but skipping due to macro changes.")
```

Comment:
Would it be possible to include the name of the changed/added/deleted macro in this log message? I'm thinking something like:
```
Change detected to macro generate_schema_name. Starting full parse.
```
Or if we cannot tell:
```
Change detected to override macro used during parsing. Starting full parse.
```

Reply:
I think this is also the right spot to fire off a `track_partial_parser` event with a new `ReparseReason` (something like `special_override_macro`)

Reply:
Technically we don't actually know that a particular macro has changed, we know that the file that contains that macro has changed. I'm a little afraid that if we specify the macro by name we'd get people complaining that they hadn't changed that macro... Should I put the macro file name in the message instead?

Reply:
Got it. I like the idea of including the file name. This sounds a bit clunky, but it feels closest to the meaning we want to get across:
```
Change detected in {file_name}, which defines a macro widely used in parsing. Starting full parse.
```